pkg/registry/core/pod/storage/eviction.go

```diff
@@ -331,6 +332,8 @@ func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb p
 	}
 
 	pdb.Status.DisruptionsAllowed--
+	pdbhelper.UpdateDisruptionAllowedCondition(&pdb)
```

Comment:
only need to update this if `pdb.Status.DisruptionsAllowed` is now 0

skipping this otherwise leaves the ObservedGeneration from the controller in place, which is helpful since that reflects the label selector the controller used to compute the CurrentHealthy/DesiredHealthy status fields as well